core/src/main/java/org/radargun/stages/test/legacy/LegacyStressor.java

```diff
@@ -87,15 +91,18 @@ private void runInternal() {
                if (ongoingTx != null) {
                   endTransactionAndRegisterStats(null);
                }
-               stats.begin();
-               this.started = started;
                break;
-            }
-            Operation operation = operationSelector.next(random);
-            try {
-               logic.run(operation);
-            } catch (OperationLogic.RequestException e) {
-               // the exception was already logged in makeRequest
+            } else {
```

Comment:
I'll reformulate the problems:
1) Until all other threads are started (`stage.isStarted()` is true), the thread should execute operations in a loop (though these are not recorded). Your change executes only single operation before that.
2) `stats.begin()` should be called as late as possible, before starting to execute recorded operations. Your change calls that right after first operation, before the other threads start (`stage.isStarted()` returns true).